frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card-item.tsx

```diff
@@ -1,75 +1,114 @@
 
 const formatCount = (count: number) => {
   const hCount = humanizeNumber(count);
   return `${hCount.string} Object${count === 1 ? '' : 's'}`;
 };
 
-const BucketsRowStatus: React.FC<BucketsRowStatusProps> = React.memo(({ status, link, error }) => (
-  <div className="nb-buckets-card__row-status-item">
-    {error || _.isNil(status) ? (
-      <span className="co-dashboard-text--small nb-buckets-card__row-subtitle">Not available</span>
-    ) : Number(status) > 0 ? (
-      <>
-        <a href={link} style={{ textDecoration: 'none' }} target="_blank" rel="noopener noreferrer">
-          <RedExclamationCircleIcon className="co-dashboard-icon nb-bucket-card__status-icon" />
-          <span className="nb-buckets-card__row-status-item-text">{status}</span>
-        </a>
-      </>
-    ) : null}
-  </div>
-));
+// Displays count of erroneous buckets due to issues not externalized as phase
+const OtherFailure: React.FC<BucketFailureItemProps> = React.memo(({ link, status }) => {
+  return (
+    <div className="nb-buckets-card__buckets-failure-status-item">
+      <a href={link} style={{ textDecoration: 'none' }} target="_blank" rel="noopener noreferrer">
+        <RedExclamationCircleIcon />
+        <span className="nb-buckets-card__buckets-failure-status-count">{status}</span>
+      </a>
+    </div>
+  );
+});
 
-const BucketsRow: React.FC<BucketsRowProps> = React.memo(
-  ({ bucketsCount, title, objectsCount }) => {
-    const subtitle: string = _.isNil(objectsCount)
-      ? 'Not available'
-      : formatCount(Number(objectsCount));
-    return (
-      <div className="nb-buckets-card__row-title">
-        <div>{_.isNil(bucketsCount) ? title : pluralize(Number(bucketsCount), title)}</div>
-        <div className="co-dashboard-text--small nb-buckets-card__row-subtitle">{subtitle}</div>
-      </div>
-    );
+// Displays count of erroneous buckets due to failure in provision
+const ProvisioningFailure: React.FC<BucketFailureItemProps> = React.memo(({ link, status }) => {
+  return (
+    <div className="nb-buckets-card__buckets-failure-status-item">
+      <Link to={link} style={{ textDecoration: 'none' }}>
+        <InProgressIcon className="co-inventory-card__status-icon--progress" />
+        <span className="nb-buckets-card__buckets-failure-status-count">{status}</span>
+      </Link>
+    </div>
+  );
+});
+
+// Displays count of erroneous buckets
+const BucketFailureStatus: React.FC<{ count: number[]; link: string[] }> = React.memo(
+  ({ count, link }) => (
+    <div className="nb-buckets-card__buckets-failure-status">
+      {count[0] > 0 && <OtherFailure link={link[0]} status={count[0]} />}
+      {count[1] > 0 && <ProvisioningFailure link={link[1]} status={count[1]} />}
+    </div>
+  ),
+);
+
+// Displays count of buckets and objects present in buckets
+const BucketsStatus: React.FC<BucketsStatusProps> = React.memo(
+  ({ isLoading, hasLoadError, title, bucketsCount, objectsCount }) => {
+    let body: JSX.Element;
```

Comment:
```suggestion
    const body = (isLoading && !hasLoadErrors) ? 
    (
    <>
          <div className="co-inventory-card__item-title">
            <div className="skeleton-inventory" />
            <div>{title}</div>
          </div>
          <div className="skeleton-text" />
        </>
        ) : (
          <>
          <div>{hasLoadError ? title : pluralize(bucketsCount, title)}</div>
          <div className="co-dashboard-text--small text-secondary">
            {hasLoadError || !objectsCount ? 'Not available' : formatCount(Number(objectsCount))}
          </div>
        </>
        )
```

Reply:
Whats wrong with if else ? That looks cleaner and non cryptic too.

Reply:
not a problem with if-else. Use consts over let whenever possible

Reply:
Well its said because its good to declare const when you dont have a variable used anywhere or declared since a let is `undefined`. I am aware of this.
Here it looks good, we wont run into any issues, the variable `body` will not be undefined.

Reply:
if it's not being reused then IMO `const` is a better candidate. After all, the ternary operator is more than capable of handling since the logic is not that complex.